src/PaymentGateways/Gateways/PayPalStandard/Controllers/PayPalStandardWebhook.php

```diff
@@ -28,23 +29,31 @@ public function __construct(WebhookValidator $webhookValidator)
      * Handle PayPal ipn
      *
      * @since 2.19.0
+     * @unreleased Respond with 200 http status to ipn.
      */
     public function handle()
     {
         $eventData = file_get_contents('php://input');
         $eventData = wp_parse_args($eventData);
 
-        if ( ! $this->webhookValidator->verifyEventSignature($eventData)) {
-            wp_die('Forbidden', 404);
+        if (!$this->webhookValidator->verifyEventSignature($eventData)) {
+            exit();
         }
 
         $donationId = isset($eventData['custom']) ? absint($eventData['custom']) : 0;
         $txnType = $eventData['txn_type'];
 
         // ipn verification can be disabled in GiveWP (<=2.15.0).
         // This check will prevent anonymous requests from editing donation, if ipn verification disabled.
-        if ( ! $this->verifyDonationId($donationId)) {
-            wp_die('Forbidden', 404);
+        if (!$this->verifyDonationId($donationId)) {
+            Log::info(
+                'PayPal Standard IPN Error',
```

Comment:
I want to be clear on this. _Is_ this an error? Or is it quite possible that GiveWP could receive events from PayPal that it simply isn't concerned with?

Reply:
fixed: https://github.com/impress-org/givewp/compare/640467c6a732...2057d9cfadd8

Reply:
Sorry if I was unclear. I wasn't suggesting a change. I'm genuinely asking whether this scenario is an error, or if it's legitimately possible for PayPal to send an event that's not associated with a donation on GiveWP?

Reply:
@JasonTheAdams yes, your statement is true, and that is why it is an error.

Reply:
Sorry man, I feel like we're crossing signals here. What, specifically is true?
1. It is completely normal for PayPal to send an event that GiveWP doesn't have a donation for — therefore this **is not** an error
2. IPN doesn't work that way and all incoming events should be unique to GiveWP — therefore this **is** an error.

Reply:
@JasonTheAdams Currently we are handling ipn only for donations.

Here are scenarios:
- complete a donation for successful payment.
- refund a donation for refunded/reversed payment.
- add a note to donation for pending payment.

As you can see we need a donation id in each scenario.
For these reasons, I am adding log as an error if the donation id is missing.

Let me know if you have any questions.